test/core/gprpp/time_util_test.cc

```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <gtest/gtest.h>
+
+#include <grpc/support/time.h>
+
+#include "absl/time/time.h"
+
+#include "src/core/lib/gprpp/time_util.h"
+
+TEST(TimeUtilTest, ToGprTimeSpecFromAbslDuration) {
+  EXPECT_EQ(0,
+            gpr_time_cmp(gpr_inf_future(GPR_TIMESPAN),
+                         grpc_core::ToGprTimeSpec(absl::InfiniteDuration())));
+  EXPECT_EQ(0,
+            gpr_time_cmp(gpr_inf_past(GPR_TIMESPAN),
+                         grpc_core::ToGprTimeSpec(-absl::InfiniteDuration())));
+  EXPECT_EQ(0, gpr_time_cmp(gpr_time_0(GPR_TIMESPAN),
+                            grpc_core::ToGprTimeSpec(absl::ZeroDuration())));
+
+  std::vector<int> times = {-10, -1, 0, 1, 10};
```

Comment:
Please avoid unnecessary loop logic in tests.  Let's just put each of these checks in its own test, with expectations that are hard-coded instead of computed.  That will ultimately make the tests more robust.

Reply:
Can you give me an example about how to change? I though this was clear enough in despite of loop-logic. Without this, it might have to get a loop unrolling, which is against readability. Also expected values are sort of hard-coded values although they look computed.

Reply:
Hmm, sorry, I think I misunderstood this code when I first looked at it.  I thought each one was a different test, but I see that you're doing different units for each one (e.g., minutes, hours, etc).  This is fine.